core/field_input.ts

```diff
@@ -642,13 +644,13 @@ export abstract class FieldInput<T extends InputTypes> extends Field<
     // rendered blocks.
     if (!(block instanceof BlockSvg)) return false;
 
-    bumpObjects.bumpIntoBounds(
+    const bumped = bumpObjects.bumpIntoBounds(
       this.workspace_!,
       this.workspace_!.getMetricsManager().getViewMetrics(true),
       block,
     );
 
-    this.resizeEditor_();
+    if (!bumped) this.resizeEditor_();
```

Comment:
Why do you only need to resize the editor if it wasn't bumped? Should it be the opposite?

Reply:
If we trigger a move of the block (the block gets bumped), it triggers `onLocationChange` for all fields, so the field gets bumped that way. If we don't move the block, we still need to reposition the widget because the block might be moving relative to the top-left of the window (even though it's not moving relative to the workspace origin).

Here's what it looks like if I remove the `if (!bumped) this.resizeEditor_();` line:
![Screen recording 2023-10-26 9 31 51 AM](https://github.com/google/blockly/assets/25440652/0ea82227-10f9-435c-9a21-716c579c321b)

You can see that if we do bump, repositioning is handled, but if we don't it is not handled.